stl/inc/functional

```diff
@@ -1207,6 +1207,604 @@ _NODISCARD bool operator!=(nullptr_t, const function<_Fty>& _Other) noexcept {
 }
 #endif // !_HAS_CXX20
 
+#if _HAS_CXX23
+union alignas(max_align_t) _Move_only_function_data {
+    void* _Pointers[_Small_object_num_ptrs];
+    const void* _Impl;
+    char _Data; // For aliasing

```

Comment:
Probably not by standard, but otherwise the trick works even with other compiler. I'll see if we can avoid unnecessary aliasing

Reply:
Strict Aliasing Rule is [[expr.basic.lval]/11](https://eel.is/c++draft/expr#basic.lval-11). It says:

> If a program attempts to access the stored value of an object through a glvalue whose type is not similar to one of the following types the behavior is undefined:
>  * the dynamic type of the object,

_Similar_ is defined in [[conv.qual]/2](https://eel.is/c++draft/expr#conv.qual-2):

> Two types T1 and T2 are similar if they have qualification-decompositions with the same n such that corresponding P components are either the same or one is “array of N” and the other is “array of unknown bound of”, and the types denoted by U are the same.

_Qualification decomposition_ is defined just above in ([conv.qual]/1])[https://eel.is/c++draft/expr#conv.qual-1].

When we apply it, `void*` and `const void*` are _similar types. 

----

Then about _Data. Back to [[expr.basic.lval]/11](https://eel.is/c++draft/expr#basic.lval-11). 11.3 says:
> a char, unsigned char, or std​::​byte type.

----

So nothing here is breaking the rules.
I mean tried to avoid `const void*` and `void*[N]` union, and it turned out to be very inconvenient.
I can try to get rid of `_Data` if you hate it.

Reply:
I think this is fine as-is, thanks for explaining!